api/task_dag.go

```diff
@@ -30,8 +30,9 @@ type TaskDAGFind struct {
 }
 
 // TaskIndexDAG describes task dependency relationship using array index to represent task.
-// It is needed because we don't know task id before insertion.
+// It is needed because we don't know task id before insertion, so we describe the dependency
+// using the in-memory representation, i.e, the array index.
 type TaskIndexDAG struct {
-	FromIndex int
```

Comment:
Should be Index to better indicate that's the index than ID.

Reply:
I thought it's already conveyed in the type name `TaskIndexDAG` and the comments of the struct.

Reply:
Not everyone reads comments and it's always preferred to let the name tell the story.

Reply:
SG. Reverted.